ROADMAP.md

```diff
@@ -1,24 +1,72 @@
 
-This document summarizes the overall development plans for PlasmaPy.
-This roadmap is very fluid and is subject to change due to community
-priorities.  For the most up-to-date information, see PlasmaPy's
-[issues](https://github.com/PlasmaPy/plasmapy/issues), [pull
-requests](https://github.com/PlasmaPy/plasmapy/pulls), and
-[projects](https://github.com/PlasmaPy/PlasmaPy/projects).
+This document summarizes the long-term development plans for PlasmaPy.
+This roadmap is very fluid and will continue to be updated and changed
+throughout the development process based on community input.  
 
-## Features to be included in Version 0.1 release
+If you have feature requests, we highly encourage you to [raise an
+issue](https://github.com/PlasmaPy/PlasmaPy/issues/new) in [PlasmaPy's
+GitHub repository](https://github.com/PlasmaPy/PlasmaPy).  More
+in-depth development plans are contained within the repository for
+[PlasmaPy Enhancement Proposals
+(PLEPs)](https://github.com/PlasmaPy/PlasmaPy-PLEPs) for some topics.
+Additional resources (including grant proposals) may found on the
+[PlasmaPy Community on
+Zenodo](https://zenodo.org/communities/plasmapy).
 
-This is to be a preview/prototype release.
+## Objective
+
+The goal of PlasmaPy is to foster a fully open source software
+ecosystem for plasma research and education.  Functionality needed by
+most plasma physicists will go in the PlasmaPy core package, whereas
+more specialized functionality will go into PlasmaPy's affiliated
+packages.
+
+## Code development priorities
+
+### High level data structures
+
+One of the most fundamental design decisions for PlasmaPy is how to
+represent different plasmas.  This decision is challenging given that
+plasma data comes from many different sources, in many different
+formats, and with many different conventions for storing and
+representing metadata.  Creating a single base class will become
+monolithic very quickly, whereas creating numerous different classes
+has the potential problems of requiring users to remember many
+different classes with diverging application programming interfaces
+(APIs).  PlasmaPy is taking a hybrid approach (described in [PLEP
+6](http://doi.org/10.5281/zenodo.1460977)) that involves creating an
+[abstract base class
+(ABC)](https://docs.python.org/3/library/abc.html) that contains
+abstract methods that have to be defined for each of the subclasses
+for different plasma representations (thus ensuring a more consistent
+API).  Users can then invoke the `Plasma` class factory to select and
+instantiate the appropriate subclass based on inputs provided by the
+user.  The framework for base data structures was implemented by
+Google Summer of Code student Ritiek Malhotra in 2018, along with the
+[OpenPMD](https://github.com/openPMD/openPMD-standard) subclass.  The
+next major development tasks are to create subclasses for different
+plasma representations and to expand/refine the API defined by the
+ABC.  
```

Comment:
Reading this now, an idea comes to mind: perhaps we could say that for 0.3 we want to have an API design document for the `Plasma`. I don't think the ABC we have now really defines what we want to have there - it's the start of an implementation, sure, but since it's missing a lot of moving parts and access points, it doesn't explicitly say how we want that to behave.

Perhaps we could make discussing this the main point of a telecon sometime?